modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectActionLocalServiceImpl.java

```diff
@@ -439,7 +446,35 @@ private void _validateObjectActionExecutorKey(
 					"No object action executor is registered with " +
 						objectActionExecutorKey);
 			}
+
+			return;
 		}
+
+		ObjectActionExecutor objectActionExecutor =
+			_objectActionExecutorRegistry.getObjectActionExecutor(
+				objectActionExecutorKey);
+
+		if ((objectActionExecutor.getCompanyId() != 0) &&
+			(objectActionExecutor.getCompanyId() !=
+				CompanyThreadLocal.getCompanyId())) {
+
+			return;
+		}
+
+		Set<String> allowedObjectDefinitionNames =
+			objectActionExecutor.getAllowedObjectDefinitionNames();
+
+		if (allowedObjectDefinitionNames.isEmpty() ||
+			allowedObjectDefinitionNames.contains(objectDefinitionName)) {
+
+			return;
+		}
+
+		throw new ObjectActionExecutorKeyException(
+			StringBundler.concat(
+				"The object action executor key ", objectActionExecutorKey,
+				" is not allowed for object definition ",
+				objectDefinitionName));
```

Comment:
@murilostodolni 

create a method to check is the scope criteria is met and use it to avoid duplications:

```java
ObjectActionExecutorUtil.isScopeCriteriaMet() {

if ((objectActionExecutor.getCompanyId() != 0) &&
	(objectActionExecutor.getCompanyId() !=
		CompanyThreadLocal.getCompanyId())) {

	return false;
}

Set<String> allowedObjectDefinitionNames =
	objectActionExecutor.getAllowedObjectDefinitionNames();

return allowedObjectDefinitionNames.isEmpty() ||
	   allowedObjectDefinitionNames.contains(
		   objectDefinitionName);
}
````